### ISSUE TITLE
Add `index.html` in root of Rust docs generated website

### ISSUE THREAD

**Issue by MexicanAce** [matter-labs/era-test-node#82]
#### 📝 Description

Running `cargo doc --no-deps` generates the docs website we use for the `GitHub Page` of this repo, but it never generates a starting file (e.g. `./target/docs/index.html` doesn't exist). We should update the [docs GitHub workflow](https://github.com/matter-labs/era-test-node/blob/main/.github/workflows/docs.yaml) to generate/add an `index.html` that redirects to `./era_test_node/index.html`

Once this is done, we can also update the website linked in the Project Settings of this repo.

#### 🤔 Rationale

This makes the default landing page for GitHub Pages work without throwing a 404:
https://matter-labs.github.io/era-test-node/

#### 🖼️ Mockups/Examples

New `index.html`:

```html
<!DOCTYPE HTML>
<html lang="en-US">
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="refresh" content="0; url=./era_test_node/index.html">
        <script type="text/javascript">
            window.location.href = "./era_test_node/index.html"
        </script>
        <title>Page Redirection</title>
    </head>
    <body>
        If you are not redirected automatically, follow this <a href='./era_test_node/index.html'>link to example</a>.
    </body>
</html>
```

<sub>[DEVRL-275](https://linear.app/matterlabs/issue/DEVRL-275/add-indexhtml-in-root-of-rust-docs-generated-website)</sub>

**Comment by AadiNir** [comment]
Hey I'd liked to be assigned to the issue

**Comment by MexicanAce** [comment]
Of course! 👍

**Comment by MexicanAce** [comment]
This was addressed in #107 
<img width="342" alt="image" src="https://github.com/matter-labs/era-test-node/assets/1890113/6e0ad6d9-4e57-426f-b969-acb7c84def7d">
